File: nathan/Problema0258/problema0258.py
qVoos = int(input("Quantidade de voos oferecidos: "))
vN = []
vOrigem = []
vDestino = []
vLugares = []

for i in range(0, qVoos):
    n = int(input("Número do voo %d:" %(i+1)))
    vN.append(n)
    n = input("Origem do voo %d:" %(i+1))
    vOrigem.append(n)
    n = input("Destino do voo %d:" %(i+1))
    vDestino.append(n)
    n = int(input("Lugares disponíveis do voo %d:" %(i+1)))
    vLugares.append(n)

resposta = int(input("\nSELECIONE UMA OPÇÃO \n1- Consultar; \n2- Efetuar Reserva; \n3- Sair.\n"))

while resposta != 3:
    if resposta == 1:
        consulta = int(input("\nSELECIONE UMA CONSULTA \n1 - Pelo número do voo; \n2 - Pela origem; \n3 - Por destino."))
        if consulta == 1:
            nVoo = int(input("Digite o numero do voo: "))
            for i in range(0, qVoos):
                if vN[i] == nVoo:
                    print("Numero do voo: %d" %nVoo)
                    print("Origem do voo: ", vOrigem[i])
                    print("Destino do voo: ", vDestino[i])
                    print("Lugares disponíveis no voo: %d" %vLugares[i])
        elif consulta == 2:
            origem = input("Digite a origem : ")
            if vOrigem[i] == origem:
                print("Numero do voo: %d" %vN[i])
                print("Origem do voo: ", origem)
                print("Destino do voo: ", vDestino[i])
                print("Lugares disponíveis no voo: %d" %vLugares[i])
        elif consulta == 3:
            destino = input("Digite o destino: ")
            if vOrigem[i] == origem:
                print("Numero do voo: %d" %vN[i])
                print("Origem do voo: ", vOrigem[i])
                print("Destino do voo: " ,destino)
                print("Lugares disponíveis no voo: %d" %vLugares[i])
    elif resposta == 2:
        vooReserva = int(input("Numero do voo para a reserva: "))
        vooReservaB = False
        for i in range(0, qVoos):
            if vN[i] == vooReserva:
                vooReservaB = True
                if vLugares[i] > 0:
                    print("Reserva confirmada.")
                    vLugares[i] = vLugares[i] - 1
                else:
                    print("Voo lotado.")
        if vooReservaB == False:
            print("Voo inexistente.");    

    resposta = int(input("\nSELECIONE UMA OPÇÃO \n1- Consultar; \n2- Efetuar Reserva; \n3- Sair."))            

File: nathan/Problema0258/problema0258.java
package Problema0258;

import java.util.Scanner;

public class problema0258{
    public static void main(String[] args){
        Scanner input = new Scanner(System.in);

        System.out.println("Quantidade de voos oferecidos: ");
        int qVoos = Integer.valueOf(input.nextLine());

        int vN[] = new int[qVoos];
        String vOrigem[] = new String[qVoos];
        String vDestino[] = new String[qVoos];
        int vLugares[] = new int[qVoos];

        for(int i = 0; i < qVoos; i++){
            System.out.println("Numero do voo " + (i+1) + ": ");
            vN[i] = Integer.valueOf(input.nextLine());
            System.out.println("Origem do voo " + (i+1) + ": ");
            vOrigem[i] = input.nextLine();
            System.out.println("Destino do voo " + (i+1) + ": ");
            vDestino[i] = input.nextLine();
            System.out.println("Lugares do voo " + (i+1) + ": ");
            vLugares[i] = Integer.valueOf(input.nextLine());
        }

        System.out.println("\nSELECIONE UMA OPÇÃO \n1- Consultar; \n2- Efetuar Reserva; \n3- Sair.");
        int resposta = Integer.valueOf(input.nextLine());

        while(resposta != 3){
            if(resposta == 1){
                System.out.println("\nSELECIONE UMA CONSULTA \n1 - Pelo número do voo; \n2 - Pela origem; \n3 - Por destino.");
                int consulta = Integer.valueOf(input.nextLine());
                switch(consulta){
                    case 1:
                        System.out.println("Digite o numero do voo: ");
                        int nVoo = Integer.valueOf(input.nextLine());
                        for(int i = 0; i < qVoos; i++){
                            if(vN[i] == nVoo){
                                System.out.println("Numero do voo: " + nVoo);
                                System.out.println("Origem do voo: " + vOrigem[i]);
                                System.out.println("Destino do voo: " + vDestino[i]);
                                System.out.println("Lugares disponíveis: " + vLugares[i]);
                            }
                        }
                        break;
                    case 2:
                        System.out.println("Digite a origem: ");
                        String origem = input.nextLine();
                        for(int i = 0; i < qVoos; i++){
                            if(origem.equalsIgnoreCase(vOrigem[i])){
                                System.out.println("\nNumero do voo: " + vN[i]);
                                System.out.println("Origem do voo: " + origem);
                                System.out.println("Destino do voo: " + vDestino[i]);
                                System.out.println("Lugares disponíveis: " + vLugares[i]);
                            }
                        }
                        break;
                    case 3:
                        System.out.println("Digite o destino: ");
                        String destino = input.nextLine();
                        for(int i = 0; i < qVoos; i++){
                            if(destino.equalsIgnoreCase(vDestino[i])){
                                System.out.println("\nNumero do voo: " + vN[i]);
                                System.out.println("Origem do voo: " + vOrigem[i]);
                                System.out.println("Destino do voo: " + destino);
                                System.out.println("Lugares disponíveis: " + vLugares[i]);
                            }
                        }
                        break;
                }
            }else if(resposta == 2){
                System.out.println("Numero do voo para a reserva: ");
                int vooReserva = Integer.valueOf(input.nextLine());
                boolean vooReservaB = false;
                for(int i = 0; i < qVoos; i++){
                    if(vN[i] == vooReserva){
                        vooReservaB = true;
                        if(vLugares[i] > 0){
                            System.out.println("Reserva confirmada.");
                            vLugares[i]--;
                        }else{
                            System.out.println("Voo lotado.");
                        }
                    }
                }
                if(vooReservaB == false){
                    System.out.println("Voo inexistente.");
                }
            }

            System.out.println("\nSELECIONE UMA OPÇÃO \n1- Consultar; \n2- Efetuar Reserva; \n3- Sair.");
            resposta = Integer.valueOf(input.nextLine());
        }
    }
}